rs/rust_canisters/xnet_test/src/main.rs

```diff
@@ -202,96 +181,103 @@ fn fanout() {
             continue;
         }
 
-        if canisters.contains(&self_id.get().as_slice().to_vec()) {
+        if canisters.contains(&self_id) {
             // Same subnet
             continue;
         }
 
+        let mut futures = vec![];
         for _ in 0..PER_SUBNET_RATE.with(|r| *r.borrow()) {
             let idx = RNG.with(|rng| rng.borrow_mut().gen_range(0..canisters.len()));
             let canister = canisters[idx].clone();
 
             let seq_no = STATE.with(|s| s.borrow_mut().next_out_seq_no(canister.clone()));
 
-            let msg = candid_encode_padded(&Request {
+            let payload_size = PAYLOAD_SIZE.with(|p| *p.borrow()) as usize;
+            let payload = Request {
                 seq_no,
                 time_nanos: time_nanos(),
-            });
+                padding: vec![0; payload_size.saturating_sub(16)],
+            };
 
-            let err_code = api::call_raw(
-                api::CanisterId::try_from(canister.clone()).unwrap(),
+            let res = call::<(Request,), (Reply,)>(
+                Principal::try_from(canister.clone()).unwrap(),
                 "handle_request",
-                &msg[..],
-                on_reply,
-                on_reject,
-                None,
-                std::ptr::null_mut(),
-                api::Funds::zero(),
+                (payload,),
             );
+            futures.push(res);
+        }
 
-            if err_code != 0 {
-                log(&format!(
-                    "{} call failed with {}",
-                    time_nanos() / 1_000_000,
-                    err_code
-                ));
-                METRICS.with(|m| m.borrow_mut().call_errors += 1);
-            } else {
-                METRICS.with(move |m| m.borrow_mut().requests_sent += 1);
+        let results = join_all(futures).await;
+
+        for res in results {
+            match res {
+                Ok((reply,)) => {
+                    let elapsed = Duration::from_nanos(time_nanos() - reply.time_nanos);
+                    METRICS.with(move |m| m.borrow_mut().requests_sent += 1);
```

Comment:
I think the `calls_made` approach makes the most sense. I'll try to switch to that.

Reply:
That doesn't sound right. How could `call_errors` (counting `Err` responses) be greater than `calls_made` (counting futures)?

Reply:
BTW, by "change the meaning of this metric" I meant incrementing it whenever we create a future. As opposed to on every successful response (currently) or on every successful `call_perform` (previously).

Reply:
Ah ok, I misunderstood that. Yeah I can also do that. Maybe even name it `calls_attempted`. Then `calls_attempted` - `call_errors` should really be the successful ones.

Reply:
> That doesn't sound right. How could call_errors (counting Err responses) be greater than calls_made (counting futures)?

Because I wasn't counting futures. So actually, synchronous errors can be higher as they will be updated faster compared to when an actual response comes back.

Reply:
Might even need to use `calls_attempted - call_errors - reject_responses` in some places. You could add a method or two to `Metrics` so you don't have to copy-paste this everywhere.